internal/access/credential.go

```diff
@@ -123,27 +139,33 @@ func hasMinimumCount(min int, password string, check func(rune) bool) bool {
 func checkPasswordRequirements(db *gorm.DB, password string) error {
 	settings, err := data.GetSettings(db)
 	if err != nil {
+		logging.Debugf("error while getting settings")
```

Comment:
usually we log or return, but not both. you would be better off wrapping some context into the next error line.. 
`  return fmt.Errorf("getting settings: %w", err)`